handler.go

```diff
@@ -401,18 +456,70 @@ func (fm *Frontman) processInputContinious(inputFilePath string, local bool, int
 	}
 }
 
+// processInput processes the whole list of checks in input
 // local is false if check originated from a remote node
-func (fm *Frontman) processInput(input *Input, local bool, resultsChan *chan Result) {
-	wg := sync.WaitGroup{}
+func (fm *Frontman) processInput(checks []Check, local bool, resultsChan *chan Result) {
 	startedAt := time.Now()
 
-	succeed := runServiceChecks(fm, &wg, local, resultsChan, input.ServiceChecks)
-	succeed += runWebChecks(fm, &wg, local, resultsChan, input.WebChecks)
-	succeed += runSNMPChecks(fm, &wg, local, resultsChan, input.SNMPChecks)
-
-	wg.Wait()
+	succeed := fm.runChecks(checks, resultsChan, local)
 
-	totChecks := len(input.ServiceChecks) + len(input.WebChecks) + len(input.SNMPChecks)
+	totChecks := len(checks)
 	fm.Stats.ChecksPerformedTotal += uint64(totChecks)
 	logrus.Infof("%d/%d checks succeed in %.1f sec", succeed, totChecks, time.Since(startedAt).Seconds())
 }
+
+// runs all checks in checkList and sends results to resultsChan
+func (fm *Frontman) runChecks(checkList []Check, resultsChan *chan Result, local bool) int {
+	wg := &sync.WaitGroup{}
+
+	succeed := 0
+	for _, check := range checkList {
+		wg.Add(1)
+		go func(wg *sync.WaitGroup, check Check, resultsChan *chan Result) {
+			defer wg.Done()
+
+			res, err := fm.runCheck(check, local)
```

Comment:
err is not handled?

Reply:
Err is logged in fm.runCheck() Just used here to count successes